common/dynamicconfig/file_based_client.go

```diff
@@ -177,9 +198,17 @@ func (fc *fileBasedClient) Update() error {
 
 	prev := fc.values.Swap(newValues)
 	oldValues, _ := prev.(configValueMap)
-	fc.logDiff(oldValues, newValues)
+	changedMap := fc.diffAndLog(oldValues, newValues)
 	fc.logger.Info("Updated dynamic config")
 
+	fc.subscriptionLock.Lock()
+	subscriptions := maps.Values(fc.subscriptions)
```

Comment:
Maybe add a comment saying the number of subscriptions is expected to be small enough that it's okay to create this slice on every change?

Reply:
right, there should generally be one per server role in the process, plus one in the top-level, so <= 5

Reply:
though even if it was kept as a slice, we'd still have to clone it to avoid holding subscriptionLock while calling the callbacks. so I don't think a comment adds anything.